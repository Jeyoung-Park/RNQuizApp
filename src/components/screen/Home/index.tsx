/* eslint-disable @typescript-eslint/no-use-before-define */
import * as React from 'react';
import {
	Text,
	View,
	StyleSheet,
	// Button,
	TouchableOpacity,
	SafeAreaView,
} from 'react-native';
import { useDispatch } from 'react-redux';
import { navigateToQuiz } from '../../../shared/NavigateTo';

interface HomeProps {
	// route:any,
	navigation: any;
}

const Home = ({ navigation }: HomeProps) => {
	const dispatch = useDispatch();

	const goToQuiz = () => {
		// console.log("퀴즈 풀기 버튼 클릭");
		// navigation.navigate('Quiz');
		navigateToQuiz({
			navigation,
			dispatch,
			number: 2,
		});
	};

	const goToWrongQuiz = () => {
		navigation.navigate('WrongQuiz');
	};

	return (
		// eslint-disable-next-line @typescript-eslint/no-use-before-define
		<SafeAreaView style={styles.container}>
			<View style={styles.textContainer}>
				<Text style={styles.titleText}>React Native Quiz App</Text>
				<Text style={styles.descriptionText}>
					React Native로 제작된 퀴즈 앱입니다.
				</Text>
			</View>

			<View style={styles.buttonContainer}>
				<TouchableOpacity style={styles.quizButton} onPress={goToQuiz}>
					<Text style={styles.quizButtonText}>퀴즈 풀기</Text>
				</TouchableOpacity>
				<TouchableOpacity style={styles.quizButton} onPress={goToWrongQuiz}>
					<Text style={styles.quizButtonText}>오답 노트</Text>
				</TouchableOpacity>
			</View>
		</SafeAreaView>
	);
};

export default Home;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		// justifyContent: 'space-around',
		alignItems: 'center',
	},
	textContainer: {
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
	},
	titleText: { fontSize: 32 },
	descriptionText: {
		fontSize: 20,
		marginTop: 20,
	},
	buttonContainer: {
		flex: 1,
		// backgroundColor:'red',
		width: '100%',
		justifyContent: 'space-around',
		alignItems: 'center',
		marginBottom: 50,
	},
	quizButton: {
		// flex: 1,
		width: '80%',
		paddingVertical: 20,

		backgroundColor: 'gray',

		justifyContent: 'center',
		alignItems: 'center',

		borderRadius: 20,
	},
	quizButtonText: {
		fontSize: 16,
	},
});
