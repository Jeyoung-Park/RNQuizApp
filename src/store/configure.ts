import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../models';

const configure = () => createStore(reducer, applyMiddleware(thunk));

export default configure;
