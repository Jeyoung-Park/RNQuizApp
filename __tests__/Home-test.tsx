import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../src/components/screen/Home';
import store from '../src/store';

jest.mock('react-native-simple-toast', () => ({
	SHORT: jest.fn(),
}));

it('snapshot test', () => {
	const tree = renderer
		.create(
			<Provider store={store}>
				<Home />
			</Provider>,
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
