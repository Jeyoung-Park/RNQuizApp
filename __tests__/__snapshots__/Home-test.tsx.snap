// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 32,
        }
      }
    >
      React Native Quiz App
    </Text>
    <Text
      style={
        Object {
          "fontSize": 20,
          "marginTop": 20,
        }
      }
    >
      React Native로 제작된 퀴즈 앱입니다.
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
        "marginBottom": 50,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "gray",
          "borderRadius": 20,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        퀴즈 풀기
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "gray",
          "borderRadius": 20,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 20,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        오답 노트
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
